cci/CMakeLists.txt

```diff
@@ -0,0 +1,205 @@
+        WORKING_DIRECTORY ${CCI_ROOT_PATH})
+else(WIN32)
+execute_process(COMMAND sh -c "echo ${CCI_BUILD_NUMBER}>CCI-VERSION-DIST"
+        WORKING_DIRECTORY ${CCI_ROOT_PATH})
+endif(WIN32)
+
+add_library(cascci SHARED ${CASCCI_SOURCES})
+set_target_properties(cascci PROPERTIES SOVERSION "${CCI_MAJOR_VERSION}.${CCI_MINOR_VERSION}")
+set_target_properties(cascci PROPERTIES PUBLIC_HEADER "${CCI_DIR}/cas_cci.h;${CCI_DIR}/compat_dbtran_def.h;${CCI_DIR}/broker_cas_error.h")
+target_include_directories(cascci PRIVATE ${EP_INCLUDES})
+target_link_libraries(cascci LINK_PRIVATE ${EP_LIBS} ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(cascci ${CMAKE_DL_LIBS})
+
+if(NOT WITH_CCI)
+  add_library(cascci_static STATIC ${CASCCI_SOURCES})
+  target_include_directories(cascci_static PRIVATE ${EP_INCLUDES})
+  target_link_libraries(cascci_static LINK_PRIVATE ${EP_LIBS} ${CMAKE_THREAD_LIBS_INIT})
+  target_link_libraries(cascci_static ${CMAKE_DL_LIBS})
+  set_target_properties(cascci_static PROPERTIES POSITION_INDEPENDENT_CODE ON)
+  set_target_properties(cascci_static PROPERTIES OUTPUT_NAME cas_cci)
+  if(WIN32)
+    target_compile_options(cascci_static PRIVATE 
+    "$<$<CONFIG:Debug>:/MTd>"
+    "$<$<CONFIG:MinSizeRel>:/MT>"
+    "$<$<CONFIG:Release>:/MT>"
+    "$<$<CONFIG:RelWithDebInfo>:/MT>"
+    )
+    if(TARGET_PLATFORM_BITS EQUAL 32)
+      target_compile_definitions(cascci_static PRIVATE _USE_32BIT_TIME_T)
+    endif(TARGET_PLATFORM_BITS EQUAL 32)
+    if(FOR_OTHER_DRIVER)
+      target_compile_definitions(cascci_static PRIVATE CCI_OLEDB)
+      target_compile_definitions(cascci_static PRIVATE CAS_CCI)
+    else(FOR_OTHER_DRIVER)
+      target_compile_definitions(cascci_static PRIVATE CAS_CCI_DL)
+    endif(FOR_OTHER_DRIVER)
+    set_target_properties(cascci_static PROPERTIES LINK_FLAGS_RELEASE "/NODEFAULTLIB:libcmt.lib" LINK_FLAGS_DEBUG "/NODEFAULTLIB:msvcrt.lib")
+    target_link_libraries(cascci_static LINK_PRIVATE ws2_32)
+  else(WIN32) 
+    add_dependencies(cascci_static ${EP_TARGETS})
+  endif(WIN32)
+  
+  install(TARGETS cascci_static
+    LIBRARY DESTINATION ${CUBRID_STATIC_LIBDIR} COMPONENT STATIC_CCI
+    ARCHIVE DESTINATION ${CUBRID_STATIC_LIBDIR} COMPONENT STATIC_CCI
+  )
+endif(NOT WITH_CCI)
+
+if(WIN32)
+  if(TARGET_PLATFORM_BITS EQUAL 32)
+    target_compile_definitions(cascci PRIVATE _USE_32BIT_TIME_T)
+  endif(TARGET_PLATFORM_BITS EQUAL 32)
+  if(FOR_OTHER_DRIVER)
+    target_compile_definitions(cascci PRIVATE CCI_OLEDB)
+    target_compile_definitions(cascci PRIVATE CAS_CCI)
+  else(FOR_OTHER_DRIVER)
+    target_compile_definitions(cascci PRIVATE CAS_CCI_DL)
+  endif(FOR_OTHER_DRIVER)
+  set_target_properties(cascci PROPERTIES LINK_FLAGS "/DEF:\"${CMAKE_SOURCE_DIR}/win/cascci/cascci.def\"" LINK_FLAGS_RELEASE "/NODEFAULTLIB:libcmt.lib" LINK_FLAGS_DEBUG "/NODEFAULTLIB:msvcrt.lib")
+  target_link_libraries(cascci LINK_PRIVATE ws2_32)
+endif(WIN32)
+
+if(WITH_CCI)
+add_dependencies(cascci ${EP_TARGETS})
+endif(WITH_CCI)
+
```

Comment:
I cannot understand why this dependency is required only for WITH_CCI defined case. If you select openssl build from **external** source in ../CMakeLists.txt as below, you will met error `'SSL_has_pending' was not declared`, because openssl include was not built yet, but cci sources begin to compiling.
../CMakeLists.txt:73
`set(WITH_LIBOPENSSL   "${WITH_EXTERNAL_PREFIX}" CACHE STRING "Build with openssl library (default: EXTERNAL)")
`

Reply:
That seems to be my mistake. 
My intention is ‘EP_TARGET is NULL’. 
It is modified as follows.
https://github.com/CUBRID/cubrid-cci/pull/29/commits/76a61567cde7d4fcde60b95fc309ca1c0070ddff